### ISSUE TITLE
Typescript 5.2 support

### ISSUE THREAD

**Issue by esetnik** [sequelize/sequelize#16414]
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### Describe the feature you'd like to see implemented

```
node_modules/sequelize/types/transaction.d.ts:121:13 - error TS2395: Individual declarations in merged declaration 'LOCK' must be all exported or all local
```

### Describe why you would like this feature to be added to Sequelize
The latest version of typescript 5.2 was released and sequelize is not compatible

### Is this feature dialect-specific?

- [x] No. This feature is relevant to Sequelize as a whole.
- [ ] Yes. This feature only applies to the following dialect(s): 

### Would you be willing to resolve this issue by submitting a Pull Request?

- [x] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in implementing my feature.

---

*Indicate your interest in the addition of this feature by adding the 👍 reaction. Comments such as "+1" will be removed.*

**Comment by WikiRik** [comment]
You're quick! I'm assuming this is for v6 right? Since our typing tests for `main` pass for TypeScript 5.2 (see #16412 )

**Comment by esetnik** [comment]
Yes, for v6 👍 . I see that it is already fixed in v7 alpha.

**Comment by WikiRik** [comment]
I just noticed that you're willing to submit a PR, go for it! Do also update the following line to include 4.9, 5.0, 5.1 and 5.2;
https://github.com/sequelize/sequelize/blob/e4c780cb9478d4345a1742c26be0235d00501a03/.github/workflows/ci.yml#L39

**Comment by esetnik** [comment]
@WikiRik do you plan to release more versions of v6 if I am able to fix the issue?

**Comment by WikiRik** [comment]
> @WikiRik do you plan to release more versions of v6 if I am able to fix the issue?

Yes, every merged PR towards the v6 branch triggers a release

**Comment by avifatal** [comment]
Hi,
This is my package.json:
```
 "dependencies": 
    "@nestjs/common": "10.1.3",
    "@nestjs/config": "^3.0.0",
    "@nestjs/core": "10.1.3",
    "@nestjs/sequelize": "^10.0.0",
    "sequelize-typescript": "^2.1.5"
  },
  "devDependencies": {
    "typescript": "^5.2.0"
  }
  ```
  how can I resolve for now the error...
  becuse   `"typescript": "^5.1.0"` also fails.
Thaks

**Comment by Marocco2** [comment]
> Hi, This is my package.json:
> 
> ```
>  "dependencies": 
>     "@nestjs/common": "10.1.3",
>     "@nestjs/config": "^3.0.0",
>     "@nestjs/core": "10.1.3",
>     "@nestjs/sequelize": "^10.0.0",
>     "sequelize-typescript": "^2.1.5",
>     "sequelize": "^6.32.1",
> 
>   },
>   "devDependencies": {
>     "typescript": "^5.2.0"
>   }
> ```
> 
> how can I resolve for now the error... becuse `"typescript": "^5.1.0"` also fails. Thaks

set `"typescript": "==5.1.0"`
then switch back to `"typescript": "^5.2.0"` only when issue is resolved

**Comment by abdullateef97** [comment]
> > Hi, This is my package.json:
> > ```
> >  "dependencies": 
> >     "@nestjs/common": "10.1.3",
> >     "@nestjs/config": "^3.0.0",
> >     "@nestjs/core": "10.1.3",
> >     "@nestjs/sequelize": "^10.0.0",
> >     "sequelize-typescript": "^2.1.5",
> >     "sequelize": "^6.32.1",
> > 
> >   },
> >   "devDependencies": {
> >     "typescript": "^5.2.0"
> >   }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > how can I resolve for now the error... becuse `"typescript": "^5.1.0"` also fails. Thaks
> 
> set `"typescript": "==5.1.0"` then switch back to `"typescript": "^5.2.0"` only when issue is resolved

this doesn't work

**Comment by alejoiglesias** [comment]
> > > Hi, This is my package.json:
> > > ```
> > >  "dependencies": 
> > >     "@nestjs/common": "10.1.3",
> > >     "@nestjs/config": "^3.0.0",
> > >     "@nestjs/core": "10.1.3",
> > >     "@nestjs/sequelize": "^10.0.0",
> > >     "sequelize-typescript": "^2.1.5",
> > >     "sequelize": "^6.32.1",
> > > 
> > >   },
> > >   "devDependencies": {
> > >     "typescript": "^5.2.0"
> > >   }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > how can I resolve for now the error... becuse `"typescript": "^5.1.0"` also fails. Thaks
> > 
> > 
> > set `"typescript": "==5.1.0"` then switch back to `"typescript": "^5.2.0"` only when issue is resolved
> 
> this doesn't work for me

`"typescript": "5.1.6"` works for me

**Comment by abdullateef97** [comment]
> 

thank you @alejoiglesias this works fine

**Comment by esetnik** [comment]
This should be fixed once #16442 gets merged and a new v6 release is published

**Comment by mat813** [comment]
6.33.0 fixes it for me.
